odoo/addons/base/tests/test_api.py

```diff
@@ -603,6 +603,23 @@ def test_80_filter(self):
             ps.filtered('parent_id.employee')
         )
 
+    @mute_logger('odoo.models')
+    def test_80_partition(self):
+        """ Check partition on recordsets. """
+        ps = self.partners
+        customers = ps.browse([p.id for p in ps if p.employee])
+        others = ps - customers
+
+        # partition on a single field
+        self.assertEqual(ps.partitionned(lambda p: p.employee), (customers, others))
+        self.assertEqual(ps.partitionned('employee'), (customers, others))
+
+        # partition on a sequence of fields
+        self.assertEqual(
+            ps.partitionned(lambda p: p.parent_id.employee),
+            ps.partitionned('parent_id.employee')
```

Comment:
```suggestion
        self.assertEqual(ps.partitioned(lambda p: p.employee), (customers, others))
        self.assertEqual(ps.partitioned('employee'), (customers, others))

        # partition on a sequence of fields
        self.assertEqual(
            ps.partitioned(lambda p: p.parent_id.employee),
            ps.partitioned('parent_id.employee')
```
To follow up on @MiquelRForgeFlow's suggestion